agent-inject/agent/config.go

```diff
@@ -215,9 +215,42 @@ func (a *Agent) newConfig(init bool) ([]byte, error) {
 		}
 	}
 
+	// If EnableQuit has been set, set it on the listener. If a listener hasn't
+	// been defined, set it on a new one. Also add a simple cache stanza since
+	// that's required for an agent listener.
+	if a.EnableQuit != nil {
```

Comment:
Do we need to do anything if it's false, given it defaults to false?

Reply:
Good question, and I went back and forth on this quite a bit. It would certainly be simpler to only set the enable_quit config if the annotation has been set to true, and maybe that's the best to do for now.

Currently `EnableQuit` is setup as a pointer so the config is set only if the annotation was specified (true or false), but since it defaults to false in the agent, just omitting the config setting here should be sufficient. And it's probably very unlikely that we'd ever default to enabling the quit endpoint on the vault agent side.

Reply:
Yeah I think that seems like a safe bet for now 👍

Reply:
Updated in [3b474e8](https://github.com/hashicorp/vault-k8s/pull/330/commits/3b474e85c1d97acec915e5928f435992b06515f1) to only set the config if EnableQuit is true.